### PR TITLE
Unpublished datasets

### PR DESCRIPTION
**Pull request by laurentqro**
The Legacy concept of "unpublished" is where datasets have published metadata but no data files, and deliberately so.

This PR makes sure we:

* label unpublished datasets as `Not released` on both the search results page and the dataset page
* shows different instructions to contact the publisher, or DGU, based on the presence of contact details

Part of card https://trello.com/c/Ykb4YEZO/166-display-unpublished-datasets
Design: https://projects.invisionapp.com/d/main#/console/11687113/268789175/preview

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -9,7 +9,7 @@ gem 'jbuilder', '~> 2.5'
 gem 'devise', '~> 4.3'
 gem 'devise_invitable', '~> 1.7.2'
 gem 'activeadmin', '~> 1'
-gem 'govuk_elements_rails'
+gem 'govuk_elements_rails', '~> 3.1'
```
Comment (maxf) [inline Gemfile]: Could you separate that out to a different commit to explain the change?

File: app/assets/stylesheets/dgu.scss [hunk 2]
```diff
@@ -0,0 +1,8 @@
+.dgu-secondary-text {
+  color: $secondary-text-colour;
+}
+
+.highlight {
```
Comment (maxf) [inline app/assets/stylesheets/dgu.scss]: Please use our naming convention.

File: app/assets/stylesheets/components.scss [hunk 3]
```diff
@@ -368,9 +368,3 @@ span.showHide-open-all {
     }
   }
 }
-
-
-// ==== not a component ========================================================
-.dgu-unavailable {
```
Comment (maxf) [inline app/assets/stylesheets/components.scss]: Why removed?

Comment (laurentqro) [reply app/assets/stylesheets/components.scss]: We were using `dgu-unavailable` for things that were not ... `unavailable`, so I renamed this to the more general `dgu-secondary-text`.

Moved it to the `dgu.scss` file because it's not a component.

File: Gemfile (hunk 1 above)
Comment (laurentqro) [reply Gemfile]: I like to limit gems to non-breaking subsets of versions with the use of `~>` to avoid anyone running `bundle update` and making breaking changes in our app. Ideally all our gems would do that.

http://bundler.io/v1.16/guides/updating_gems.html

File: app/assets/stylesheets/dgu.scss (hunk 2 above)
Comment (laurentqro) [reply app/assets/stylesheets/dgu.scss]: `dgu-highlight`?

File: Gemfile (hunk 1 above)
Comment (maxf) [reply Gemfile]: yes. Can you put it in a different commit though, explaining the above. Ideally doing it for all gems in one go? Just so we have a cleaner history of code changes.

File: app/assets/stylesheets/components.scss (hunk 3 above)
Comment (maxf) [reply app/assets/stylesheets/components.scss]: ok. We'll have to make it a component and try and use more semantic naming. But I'll take care of that later.

File: app/assets/stylesheets/dgu.scss (hunk 2 above)
Comment (maxf) [reply app/assets/stylesheets/dgu.scss]: yes

File: Gemfile (hunk 1 above)
Comment (laurentqro) [reply Gemfile]: ok @maxf I'll do this in a separate PR.

Comment (laurentqro) [reply Gemfile]: I removed the commit with the Gemfile changes.
